### PR TITLE
Ret 2829 Apps & Case Mgmt common – Generate email notification to Respondent when Claimant responds to an application

### PR DESCRIPTION
**Pull request by paul-peasegood**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RET-2829

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/resources/application.yaml [hunk 1]
```diff
@@ -67,7 +67,7 @@ tornado:
 
 notifications:
   citizenPortalLink: ${CITIZEN_PORTAL_LINK:https://localhost:3001/citizen-hub/}
-  govNotifyApiKey: ${GOV_UK_NOTIFY_API_KEY:AAAAA}
+  govNotifyApiKey: "etaatkey-002d2170-e381-4545-8251-5e87dab724e7-98ebded5-98be-4b30-9ea1-1dcc0b1f5157"
```
Comment (fergusoshea) [inline src/main/resources/application.yaml:70]: Was this added for debugging? I think this should be reverted

File: src/main/java/uk/gov/hmcts/reform/et/syaapi/service/NotificationService.java [hunk 2]
```diff
@@ -333,6 +312,11 @@ public void sendResponseEmailToTribunal(
             subjectLine
         );
 
+        tribunalParameters.put(
```
Comment (fergusoshea) [inline src/main/java/uk/gov/hmcts/reform/et/syaapi/service/NotificationService.java:315]: This parameter doesn't exist yet until my code is merged in.

File: src/main/java/uk/gov/hmcts/reform/et/syaapi/notification/NotificationsProperties.java [hunk 3]
```diff
@@ -65,12 +65,17 @@ public class NotificationsProperties {
...
+
     @Value("${notifications.claimantResponseNo}")
     @NotBlank
     private String claimantResponseNoTemplateId;
 
     @Value("${notifications.claimantResponseYes}")
     @NotBlank
     private String claimantResponseYesTemplateId;
+
+    @Value("${notifications.respondentResponse}")
```
Comment (fergusoshea) [inline src/main/java/uk/gov/hmcts/reform/et/syaapi/notification/NotificationsProperties.java:77]: I believe you will need to add a PMD Suppression for this file
`@SuppressWarnings("PMD.TooManyFields")`

**Review by fergusoshea — DISMISSED** [review]

File: src/test/java/uk/gov/hmcts/reform/et/syaapi/service/NotificationServiceTest.java [hunk 4]
```diff
@@ -540,5 +543,97 @@ void shouldNotSendResponseEmailToClaimantForTypeCApplication() throws Notificati
...
+        verify(notificationClient, times(1)).sendEmail(
+            any(),
+            eq("test@resRep.com"),
+            any(),
+            eq(testData.getExpectedDetails().getId().toString())
+        );
+    }
+
+    @Test
+    void shouldNotSendResponseWhenRespondentEmailDoesNotExist() throws NotificationClientException {
```
Comment (JackReevies) [inline src/test/java/uk/gov/hmcts/reform/et/syaapi/service/NotificationServiceTest.java:599]: Why do we have a `throws` on these tests?

**Review by JackReevies — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/hmcts/reform/et/syaapi/service/NotificationServiceTest.java (hunk 4 above)
Comment (paul-peasegood) [reply src/test/java/uk/gov/hmcts/reform/et/syaapi/service/NotificationServiceTest.java:599]: The send email function throws and NotificationClientException(just like other tests)

File: src/main/java/uk/gov/hmcts/reform/et/syaapi/service/NotificationService.java [hunk 5]
```diff
@@ -426,6 +403,85 @@ public void sendResponseEmailToClaimant(
         }
     }
 
+
+    /**
+     *  Send acknowledgment email to the claimant when they are responding to
+     *  an application (type A/B) made by the Respondent.
```
Comment (JackReevies) [inline src/main/java/uk/gov/hmcts/reform/et/syaapi/service/NotificationService.java:409]: very minor: two blank spaces at the start of these sentences?

File: src/main/java/uk/gov/hmcts/reform/et/syaapi/service/NotificationService.java [hunk 6]
```diff
@@ -426,6 +403,85 @@ public void sendResponseEmailToClaimant(
...
+            caseNumber,
+            subjectLine,
+            applicationType
+        );
+        respondentParameters.put(
+            HEARING_DATE,
+            hearingDate
+        );
+
+
```
Comment (JackReevies) [inline src/main/java/uk/gov/hmcts/reform/et/syaapi/service/NotificationService.java:451]: minor: double space here?

**Review by JackReevies — APPROVED** [review]

**Review by fergusoshea — APPROVED** [review]

**Review by JackReevies — APPROVED** [review]

**Pull request merged**